app/http.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+use Swoole\Atomic;
+use Swoole\Coroutine\Http\Client;
+use Utopia\Logger\Log;
+use Utopia\Logger\Logger;
+use Swoole\Coroutine\Http\Server;
+use Swoole\Http\Request as SwooleRequest;
+use Swoole\Http\Response as SwooleResponse;
+use Swoole\Runtime;
+use Swoole\Table;
+use Swoole\Timer;
+use Utopia\App;
+use Utopia\Balancing\Algorithm\Random;
+use Utopia\Balancing\Algorithm\RoundRobin;
+use Utopia\Balancing\Balancing;
+use Utopia\Balancing\Option;
+use Utopia\CLI\Console;
+use Utopia\Logger\Adapter;
+
+Runtime::enableCoroutine(true, SWOOLE_HOOK_ALL);
+
+$executorStates = new Table(1024);
+$executorStates->column('status', Swoole\Table::TYPE_STRING, 8); // 'online' or 'offline'
+$executorStates->column('state', Swoole\Table::TYPE_STRING, 16384); // State as JSON
+$executorStates->create();
+
+$roundRobinAtomic = new Atomic(0); // Only used if round-robin algo is used
+
+function markOffline(string $executorHostname, string $error, bool $forceShowError = false): void
```

Comment:
We're relying on global functions too much here. We can abstract a lot of the complexity and decouple the logic from the HTTP server with a new structure.

We can have a new internal library called `Health` under `./src/Proxy/Health/Health.php` for handling all the ping checks.

The structure could look like this:

```php
new Health()
    ->addNode(Node $node)
    ->addNode(Node $node)
    ->ping(); //ping returns an array of Nodes, each Node has the node status check result
```

We can then take all the results and store them in memory for the balancer to use when needed.

Reply:
This will help us completely remove both `markOffline`, `makeOnline` and `fetchExecutorsState`.